### PR TITLE
Move DB logic to firefly-common impl

### PR DESCRIPTION
**Pull request by peterbroadhurst**
Core half of https://github.com/hyperledger/firefly-common/pull/39

> Ready to review - but should not merge until FF Common change makes it into a release

### CODE REVIEW

**Review by nguyer — APPROVED** [review]
Just a couple of questions, but otherwise very straightforward. Looks good.

File: internal/database/sqlcommon/blob_sql_test.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func TestInsertBlobFailBegin(t *testing.T) {
 	s, mock := newMockProvider().init()
 	mock.ExpectBegin().WillReturnError(fmt.Errorf("pop"))
 	err := s.InsertBlob(context.Background(), &core.Blob{})
-	assert.Regexp(t, "FF10114", err)
+	assert.Regexp(t, "FF00175", err)
```
[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply internal/database/sqlcommon/blob_sql_test.go:87]: With our current architecture for errors, moving them between repos, means changing them - as the prefixes are reserved in each repo.

File: internal/database/sqlcommon/provider_mock_test.go [hunk 2]
```diff
@@ -72,12 +73,16 @@ func (mp *mockProvider) Name() string {
...
 
+func (mp *mockProvider) SequenceColumn() string {
+	return "seq"
+}
+
 func (mp *mockProvider) MigrationsDir() string {
 	return mp.Name()
 }
 
-func (psql *mockProvider) Features() SQLFeatures {
```
[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply internal/database/sqlcommon/provider_mock_test.go:79]: We do - we have a mix of tests that use SQLite for an E2E "real" test, like `TestBatch2EWithDB`, and others that fill out coverage by testing edge cases using the mock provider.

File: internal/database/sqlcommon/blob_sql_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply internal/database/sqlcommon/blob_sql_test.go:87]: I've added the `migration_consideration` tag so we can include it the release notes
